# Type: fix
# Subtype: I5
# English Summary: Change the position of gen.mark(callLabel)
# Chinese Summary: 更换gen.mark(callLabel)位置
# Entry Number: 6

# === Git Diff Content ===

diff --git a/src/jvm/clojure/lang/Compiler.java b/src/jvm/clojure/lang/Compiler.java
index 5f8fdc7..77dd68e 100644
--- a/src/jvm/clojure/lang/Compiler.java
+++ b/src/jvm/clojure/lang/Compiler.java
@@ -3256,12 +3256,13 @@ static class InvokeExpr implements Expr{
 			gen.ifZCmp(GeneratorAdapter.NE, onLabel);
 			}
 
-		gen.mark(callLabel); //target
 		gen.dup(); //target, target
 		gen.invokeStatic(UTIL_TYPE,Method.getMethod("Class classOf(Object)")); //target,class
 		gen.loadThis();
 		gen.swap();
 		gen.putField(objx.objtype, objx.cachedClassName(siteIndex),CLASS_TYPE); //target
+
+		gen.mark(callLabel); //target
 		objx.emitVar(gen, v);
 		gen.invokeVirtual(VAR_TYPE, Method.getMethod("Object getRawRoot()")); //target, proto-fn
 		gen.swap();
